R/profilePlot.R

```diff
@@ -134,20 +123,28 @@ profilePlot <- function(contrastDF,
         warning("plotType must be either canvasXpress or ggplot. Assigning default value 'CanvasXpress'.")
         plotType <- "canvasxpress"
     }
+
     # Make sure specified columns exist
     assertthat::assert_that(!is.null(logRatioCol),
+                            length(logRatioCol) == 1,
                             logRatioCol %in% colnames(contrastDF),
-                            msg = "logRatioCol column not found in contrastDF.")
+                            msg = "logRatioCol to be a singular value of class character and must be in contrast data.")
+
     assertthat::assert_that(!is.null(logIntCol),
+                            length(logIntCol) == 1,
                             logIntCol %in% colnames(contrastDF),
-                            msg = "logIntCol column not found in contrastDF.")
+                            msg = "logIntCol to be a singular value of class character and must be in contrast data.")
+
     assertthat::assert_that(!is.null(pvalCol),
+                            length(pvalCol) == 1,
                             pvalCol %in% colnames(contrastDF),
-                            msg = "pvalCol column not found in contrastDF.")
-    if (!missing(geneSymCol)) {
-        assertthat::assert_that(!is.null(geneSymCol),
-                                geneSymCol %in% colnames(contrastDF),
-                                msg = "geneSymCol column not found in contrastDF.")
+                            msg = "pvalCol to be a singular value of class character and must be in contrast data.")
+
+    if (!missing(geneNameCol)) {
+        assertthat::assert_that(!is.null(geneNameCol),
+                                length(geneNameCol) == 1,
+                                geneNameCol %in% names(DGEobj::getType(dgeObj, type = "geneData")[[1]]),
```

Comment:
- There must be check for the existence of geneData in the passed DGEobj
- There must be check for the geneData length

Reply:
done.